### DISCUSSION TITLE
tgcf on vps(icoso)

### DISCUSSION THREAD

**Discussion by bilal-the-dev** [aahnik/tgcf#487] [General]
i have successfully ran tgcf locally after following the readme but to run it permanently i tried it on vps called icoso, the name doesnt matter as they are all essentially same.
Now, after running sudo apt update command in my console using ssh, i created virtual environment following readme steps and after running tgcf-web it gave me link to the UI or control panel but that link doesnt seem to work whereas when i ran locally on my terminal it works.

**Comment by MobCode100** [comment]
Your vps provider must give you a public IP, and firewall enabled to port 8501, then you may able to access.

Or another way is you can try ssh port tunneling e.g. `ssh -L 8501:localhost:8501 vps_user@vps.ip`

If my answer helps, please mark it as answer. Thank you 😁.

**Reply by bilal-the-dev** [reply]
thanks for reply i thought i would be left out been working on fix it since very long
i tried logging to my vps through console by running ssh -L 8501:localhost:8501 root@myIpOfVpWhereILogin
i need to replace local host with anything and also how do i know whats my vps user, i ususally login using ssh root@ip
anyways tried with the cmd but same result the link web ui is my vps ip address

**Reply by MobCode100** [reply]
> thanks for reply i thought i would be left out been working on fix it since very long
> i tried logging to my vps through console by running ssh -L 8501:localhost:8501 root@myIpOfVpWhereILogin
> i need to replace local host with anything and also how do i know whats my vps user, i ususally login using ssh root@ip
> anyways tried with the cmd but same result the link web ui is my vps ip address

Don't replace localhost, you can just run the command, if for example your vps user is root, and the server public IP is, e.g. 10.10.134.23, then the ssh tunnel command would be

`ssh -L 8501:localhost:8501 root@10.10.134.23`

One done, you can access the web ui using http://localhost:8501 on your pc browser. 

If it helps please mark this as answer. Thank you.

**Reply by bilal-the-dev** [reply]
hi yes that worked :) i'll mark it as an answer but i am afraid i'll lose the chat, can you help me running it 24/7, i am following https://github.com/aahnik/tgcf/wiki/Using-with-systemctl but the first step is ambigous as it says just create tgcf_start.sh file in home directory, i have no home directory but root folder so i created there and ran the next step which says `chmod: cannot access 'tgcf_start.sh': No such file or directory`

**Reply by MobCode100** [reply]
Actually icoso or any vps provider, should have firewall settings to allow any port to be accessed using public ip, so once you run the tgcf-web, you can just use e.g. http://10.10.134.23:8501. But if ssh tunnel works for you then congrats. I've posted an answer regarding your question, do look at #483, but skip the first command.

**Reply by bilal-the-dev** [reply]
i cant mark your comment as answer there is no option i think you need to send answer by clicking on send as answer not sure
as for tgcf, i installed docker because of looking at first command followed the steps mentioned in how to use docker by the owner of repo, didnt work so i tried deleting the my tgcf folder in frustration, now i followed the rest steps you mentioned it seems the web ui is live but logs giving error probably because of me deleting files

**Reply by bilal-the-dev** [reply]
` tgcf.service - tgcf Telegram forwarder
     Loaded: loaded (/etc/systemd/system/tgcf.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Sun 2023-08-06 13:44:56 UTC; 13s ago
    Process: 783931 ExecStart=/venv/bin/tgcf live (code=exited, status=203/EXEC)
   Main PID: 783931 (code=exited, status=203/EXEC)
        CPU: 1ms

Aug 06 13:44:56 localhost systemd[1]: tgcf.service: Scheduled restart job, restart counter is at 5.
Aug 06 13:44:56 localhost systemd[1]: Stopped tgcf Telegram forwarder.
Aug 06 13:44:56 localhost systemd[1]: tgcf.service: Start request repeated too quickly.
Aug 06 13:44:56 localhost systemd[1]: tgcf.service: Failed with result 'exit-code'.
Aug 06 13:44:56 localhost systemd[1]: Failed to start tgcf Telegram forwarder.`

how can i start from scratch again and do i need to be in my tgcf folder as well as virtual environment while editing using nano and do i need to create .service file or an executable shell script if i follow your steps mentioned in the post. thanks a lot

**Reply by MobCode100** [reply]
Delete the old tgcf folder along it's venv, then create new folder and redo the installation steps, or delete old container, `docker container rm your_container_name`. The name can be retrieved using `docker container ls -a`

**Comment by bilal-the-dev** [comment]
this is really hard, i turned off console so did the web ui
can you break down to me what excatly i have to follow?
after installing tgcf and running it using tgcf-web, to make it work 24/7 what commands should i follow next, just the one you have shared starting from installing nano or those which the owner of repo have added such as exectuable shell file etc?

**Reply by MobCode100** [reply]
Since you're running on vps, there's no need to use docker,

After normal installation, just run the commands that I've stated on another post, but skip the first command, starts with `apt install nano` line.

**Reply by bilal-the-dev** [reply]
i did that but same error, morever in console i am getting this `channel 4: open failed: connect failed: Connection refused
channel 3: open failed: connect failed: Connection refused

**Reply by bilal-the-dev** [reply]
i just bought another vps, started from fresh same error after running systemcl status and the connection error too, by any chance can i have your social media such as discord or something i have utlized a lot of my time on it as well as money could you please help? wont take more than 10mins i'll show you live please

**Comment by 25gabriel-cmyk** [comment]
@Bilalfr123 from the conversation flow that I've observed, you are making things difficult for yourself.
It seems like you haven't fully understood the current flow, as you have jumped to another one. This discussion will never end if your questions are answered, but you doing it your own way. Try starting over with a completely fresh install VPS and i suggest using Ubuntu, to get it online 24/7 you can simply using screen session.

**Reply by bilal-the-dev** [reply]
can i have your social media brother as i can't really show much of what i have done, i just tried from fresh but it's still same error `× tgcf.service - tgcf Telegram forwarder
     Loaded: loaded (/etc/systemd/system/tgcf.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Sun 2023-08-06 15:21:20 UTC; 10s ago
    Process: 785564 ExecStart=/venv/bin/tgcf live (code=exited, status=203/EXEC)
   Main PID: 785564 (code=exited, status=203/EXEC)
        CPU: 2ms

Aug 06 15:21:20 localhost systemd[1]: tgcf.service: Scheduled restart job, restart counter is at 5.
Aug 06 15:21:20 localhost systemd[1]: Stopped tgcf Telegram forwarder.
Aug 06 15:21:20 localhost systemd[1]: tgcf.service: Start request repeated too quickly.
Aug 06 15:21:20 localhost systemd[1]: tgcf.service: Failed with result 'exit-code'.
Aug 06 15:21:20 localhost systemd[1]: Failed to start tgcf Telegram forwarder.`

**Comment by 25gabriel-cmyk** [comment]
I suggest ensuring that you successfully install and run TGCF web on your VPS before moving on to the next step of making it run 24/7.
Also, as I mentioned before, to make it run 24/7, i found it much easier to [ATTACH IT TO A SCREEN SESSION](https://github.com/aahnik/tgcf/wiki/Deploy-to-Digital-Ocean) rather than registering its service to systemd.

```
# make sure you are inside your my-tgcf directory
screen -S tgcfSession
source .venv/bin/activate
tgcf-web

```
Exit the screen session by Ctrl + a then press d. Now you can safely close the console to VPS.

**Reply by bilal-the-dev** [reply]
yep, i first tried running only tgcf-web it worked then straight after ran the three commands you mentioned, then control a and d, my screen got detached, as soon as i closed the terminal the local host went off it only works when terminal is on

**Reply by MobCode100** [reply]
> yep, i first tried running only tgcf-web it worked then straight after ran the three commands you mentioned, then control a and d, my screen got detached, as soon as i closed the terminal the local host went off it only works when terminal is on

Ssh tunneling only works when you're connected to ssh. So you can't access it on localhost anymore after disconnecting. But to run tgcf 24/7 you don't need to run the ui. Only tgcf is what you need which on the tgcf-web you can access it on the run page.

Which once you run, it'll keep running even when you can't access the tgcf-web.

**Reply by 25gabriel-cmyk** [reply]
how about accessing it through "YourVpsIP:8501" ?

**Reply by MobCode100** [reply]
> how about accessing it through "YourVpsIP:8501" ? 

@Bilalfr123 if you can't, then there is an option to enable port 8501, look for firewall settings. I've mentioned this multiple times earlier in this discussion. 

You've even bought a vps, if that vps provider can't provide you with that settings then that vps provider is bad, and you're wasting your money.

**Reply by bilal-the-dev** [reply]
well it works for now, so yeah its contabo 8gb ram vps, i dont know if there's firewall settings

**Reply by bilal-the-dev** [reply]
bro this saved me all the time this was hosted lol thank you 💘

**Comment by MobCode100** [comment]
There are a lot of ways you can run 24/7. I assumed you only want the process to run after closing the ssh connection. But before that, try to run tgcf-web, configure everything, your connections, login your telegram account, and try to run the tgcf, making sure the message forwarding works fine. 

After everything is fine, then you can start to worry how to run 24/7. @25gabriel-cmyk has shown you the easiest way to do that.

**Reply by bilal-the-dev** [reply]
yes, it works and forwards but i just realized i was trying the local host website , LOL while the vps ip address was hosted 24/7 lol all the time i was scammed, gabriel comment made me check vps ip and it was working thanks a lot guys😍💘

**Reply by bilal-the-dev** [reply]
one last thing, now my the web ui based on my vps ip will always remain online and i can edit my configuration there so need of terminal anymore and ssh tunneling right  ? also there's no option to mark as answer

**Reply by MobCode100** [reply]
Well if you can access using public ip or your vps ip, there's no need to use ssh anymore, unless the server restarts, or something goes wrong with tgcf-web, then you may need ssh to restart the tgcf-web process.

**Reply by bilal-the-dev** [reply]
Great thank you

**Comment by 25gabriel-cmyk** [comment]
> yes, it works and forwards but i just realized i was trying the local host website , LOL while the vps ip address was hosted 24/7 lol all the time i was scammed, gabriel comment made me check vps ip and it was working thanks a lot guys😍💘

Lol, that's what i've been guessing since the very beginning 🤣

**Reply by bilal-the-dev** [reply]
haha man i am so noob
